### PR TITLE
Add semantic routing to CupertinoAlertDialog

### LINKED ISSUE #21250: Cupertino dialogs do not have route semantics or lables
Each Cupertino alert dialog widget needs to include a Semantics node with scopes/names route and a label of "ALERT" (localized).

The accessibility tree currently looks like so for a cupertino alert dialog:
```
I/flutter (10104): SemanticsNode#0
I/flutter (10104):  │ Rect.fromLTRB(0.0, 0.0, 1080.0, 1794.0)
I/flutter (10104):  │
I/flutter (10104):  └─SemanticsNode#68
I/flutter (10104):    │ Rect.fromLTRB(0.0, 0.0, 411.4, 683.4) scaled by 2.6x
I/flutter (10104):    │ textDirection: ltr
I/flutter (10104):    │
I/flutter (10104):    └─SemanticsNode#113
I/flutter (10104):      │ Rect.fromLTRB(0.0, 0.0, 411.4, 683.4)
I/flutter (10104):      │ flags: scopesRoute
I/flutter (10104):      │
I/flutter (10104):      ├─SemanticsNode#114
I/flutter (10104):      │ │ Rect.fromLTRB(70.7, 297.5, 340.7, 364.5)
I/flutter (10104):      │ │ scrollExtentMin: 0.0
I/flutter (10104):      │ │ scrollPosition: 0.0
I/flutter (10104):      │ │ scrollExtentMax: 0.0
I/flutter (10104):      │ │
I/flutter (10104):      │ └─SemanticsNode#115
I/flutter (10104):      │     Rect.fromLTRB(20.0, 23.0, 250.0, 47.0)
I/flutter (10104):      │     label: "Discard draft?"
I/flutter (10104):      │     textDirection: ltr
I/flutter (10104):      │
I/flutter (10104):      └─SemanticsNode#116
I/flutter (10104):        │ Rect.fromLTRB(70.7, 364.9, 340.7, 409.9)
I/flutter (10104):        │ scrollExtentMin: 0.0
I/flutter (10104):        │ scrollPosition: 0.0
I/flutter (10104):        │ scrollExtentMax: 0.0
I/flutter (10104):        │
I/flutter (10104):        ├─SemanticsNode#117
I/flutter (10104):        │ │ Rect.fromLTRB(0.0, 0.0, 134.8, 45.0)
I/flutter (10104):        │ │ actions: tap
I/flutter (10104):        │ │
I/flutter (10104):        │ └─SemanticsNode#118
I/flutter (10104):        │     Rect.fromLTRB(0.0, 0.0, 134.8, 45.0)
I/flutter (10104):        │     actions: tap
I/flutter (10104):        │     label: "Discard"
I/flutter (10104):        │     textDirection: ltr
I/flutter (10104):        │
I/flutter (10104):        └─SemanticsNode#119
I/flutter (10104):          │ Rect.fromLTRB(135.2, 0.0, 270.0, 45.0)
I/flutter (10104):          │ actions: tap
I/flutter (10104):          │
I/flutter (10104):          └─SemanticsNode#120
I/flutter (10104):              Rect.fromLTRB(0.0, 0.0, 134.8, 45.0)
I/flutter (10104):              actions: tap
I/flutter (10104):              label: "Cancel"
I/flutter (10104):              textDirection: ltr
```

### PR DESCRIPTION
**Pull request by mklim**
Fixes #21250

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/dialog.dart [hunk 1]
```diff
@@ -218,6 +219,8 @@ class CupertinoAlertDialog extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    final CupertinoLocalizations localizations = CupertinoLocalizations.of(context)
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/cupertino/dialog.dart]: I would remove the `??`, we should be providing the default localizations anyway.

File: packages/flutter/lib/src/cupertino/localizations.dart [hunk 2]
```diff
@@ -80,6 +80,9 @@ abstract class CupertinoLocalizations {
   /// The abbreviation for post meridiem (after noon) shown in the time picker.
   String get postMeridiemAbbreviation;
 
+  /// The term used by the system to announce dialog alerts.
+  String get semanticAlertLabel;
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/cupertino/localizations.dart]: I would call this "alertDialogLabel" to match the material localizations

File: packages/flutter/test/cupertino/dialog_test.dart [hunk 3]
```diff
@@ -63,6 +64,31 @@ void main() {
...
+          CupertinoDialogAction(child: Text('OK')),
+        ],
+      )
+    )));
+
+    expect(semantics, includesNodeWith(
+      label: 'Alert',
+      flags: <SemanticsFlag>[SemanticsFlag.namesRoute, SemanticsFlag.scopesRoute]
+    ));
+    expect(semantics, includesNodeWith(label: 'The Title'));
```
Comment (jonahwilliams) [inline packages/flutter/test/cupertino/dialog_test.dart]: If you're going to test the content of multiple nodes I would just use the hasSemantics matcher

File: packages/flutter/lib/src/cupertino/dialog.dart (hunk 1 above)
Comment (mklim) [reply packages/flutter/lib/src/cupertino/dialog.dart]: It looks like `localizations` is always null. Is there any more work that needs to be done to initialize it correctly? Looks like this eventually rolls up to [`inheritFromWidgetOfExactType`](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/framework.dart#L1925), which should be callable in `build()` from what I can tell.

Comment (jonahwilliams) [reply packages/flutter/lib/src/cupertino/dialog.dart]: Ahh you are right - it looks like we do not provide it anywhere yet

File: packages/flutter/lib/src/cupertino/dialog.dart [hunk 4]
```diff
@@ -949,18 +958,21 @@ class _PressableActionButtonState extends State<_PressableActionButton> {
...
-        onTapDown: (TapDownDetails details) => setState(() {
-          _isPressed = true;
-        }),
-        onTapUp: (TapUpDetails details) => setState(() {
-          _isPressed = false;
-        }),
-        // TODO(mattcarroll): Cancel is currently triggered when user moves past slop instead of off button: https://github.com/flutter/flutter/issues/19783
-        onTapCancel: () => setState(() => _isPressed = false),
-        child: widget.child,
+      child: new MergeSemantics(
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/cupertino/dialog.dart]: Do we need this merge semantics here?

**Review by jonahwilliams — APPROVED** [review]
LGTM,

Provided you have manually verified pressing buttons w/ talkback gestures works as expected

File: packages/flutter/lib/src/cupertino/dialog.dart (hunk 4 above)
Comment (mklim) [reply packages/flutter/lib/src/cupertino/dialog.dart]: Yes, otherwise the buttons aren't focusable.

**Pull request merged**